src/BufferStream.ts

```diff
@@ -237,38 +237,39 @@ export class BufferStream {
         return hexString.toUpperCase();
     }
 
+    checkSize(step: number): void {
+        let size = this.buffer.byteLength;
+        if (this.offset + step < size) {
+            return
+        }
 
-    checkSize(step) {
-        while (this.offset + step > this.buffer.byteLength) {
-            //throw new Error("Writing exceeded the size of buffer");
-            //resize
-            var dst = new ArrayBuffer(this.buffer.byteLength * 2);
-            new Uint8Array(dst).set(new Uint8Array(this.buffer));
-            this.buffer = dst;
-            this.view = new DataView(this.buffer);
+        while (this.offset + step > size) {
+            size *= 2;
         }
+        var dst = new ArrayBuffer(size);
+        new Uint8Array(dst).set(new Uint8Array(this.buffer));
+        this.size = size
+        this.buffer = dst;
+        this.view = new DataView(this.buffer);
     }
 
-    concat(stream) {
-        var newbuf = new ArrayBuffer(this.offset + stream.size), int8 = new Uint8Array(newbuf);
-        int8.set(new Uint8Array(this.getBuffer(0, this.offset)));
+    concat(stream: BufferStream): void {
+        this.checkSize(stream.size)
+
+        const int8 = new Uint8Array(this.buffer);
         int8.set(new Uint8Array(stream.getBuffer(0, stream.size)), this.offset);
-        this.buffer = newbuf;
-        this.view = new DataView(this.buffer);
         this.offset += stream.size;
-        this.size = this.offset;
```

Comment:
We should restore the `this.size = this.offset;` here when the `this.size = size` above is removed.